$NetBSD: patch-ad,v 1.5 2026/08/04 14:59:11 ryoon Exp $

- use standard headers
- don't declare own errno
- fix build with GCC 14

--- util.c.orig	2026-08-04 12:05:46.864661412 +0000
+++ util.c
@@ -1,4 +1,7 @@
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
 #ifdef SDBM
 #include "sdbm.h"
 #else
@@ -10,15 +13,13 @@ register char *s2;
 register char *s1;
 register char *s2;
 {
-	extern int errno, sys_nerr;
-	extern char *sys_errlist[];
 	extern char *progname;
 
 	if (progname)
 		fprintf(stderr, "%s: ", progname);
 	fprintf(stderr, s1, s2);
-	if (errno > 0 && errno < sys_nerr)
-		fprintf(stderr, " (%s)", sys_errlist[errno]);
+	if (errno > 0)
+		fprintf(stderr, " (%s)", strerror(errno));
 	fprintf(stderr, "\n");
 	exit(1);
 }
@@ -28,7 +29,7 @@ char *pag;
 char *pag;
 {
 	register unsigned n;
-	register off;
+	register int off;
 	register short *ino = (short *) pag;
 
 	if ((n = ino[0]) > PBLKSIZ / sizeof(short))
